src/ray/raylet/scheduling/local_resource_manager.h

```diff
@@ -108,6 +118,9 @@ class LocalResourceManager : public syncer::ReporterInterface {
 
   void ReleaseWorkerResources(std::shared_ptr<TaskResourceInstances> task_allocation);
 
+  void SetBusyFootprint(WorkFootprint item);
```

Comment:
I actually think we should keep it this way, due to some subtle differences (2 points):

1. In the current codepath, we don't always call `OnResourceOrStateChanged()` with `SetResourceIdle/SetResourceNonIdle` However, with `SetIdleFootprint`, `SetBusyFootprint`, we will do it when the state changes. 

`OnResourceOrStateChanged()` is very important because it increments the version number for Ray Syncer, which is what will actually prompt a message to be sent the next time the reporter polls for messages. If we don't call it, other nodes (GCS and Autoscaler) won't know the idle state changed. Assuming the current logic is correct, there are differences in how and when we call it between a WorkFootprint and a Resource that will make unification messy.

2. SetResourcesIdle and SetResourcesNonIdle are private. The WorkFootprint ones are public. Why make the former public too when we have protected it from external callers?

Reply:
I see - good points.